examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjectsStorage.agent.ts

```diff
@@ -116,17 +160,21 @@ export class BpiSubjectStorageAgent extends PrismaService {
   }
 
   async getBpiSubjectByPublicKey(publicKey: string): Promise<BpiSubject> {
+    const bpiSubjectKey = await this.prisma.publicKey.findUnique({
```

Comment:
No reason for two queries. The one below is enough but the where clause should be fixed to  you just query by the publicKey.value based on a inner join with the bpiSubject table.